x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ccr/AutoFollowStats.java

```diff
@@ -116,14 +114,9 @@ public AutoFollowStats(StreamInput in) throws IOException {
         numberOfFailedFollowIndices = in.readVLong();
         numberOfFailedRemoteClusterStateRequests = in.readVLong();
         numberOfSuccessfulFollowIndices = in.readVLong();
-        // note: the casts to the following Writeable.Reader<T> instances are needed by some IDEs (e.g. Eclipse 4.8) as a compiler help
-        recentAutoFollowErrors = new TreeMap<>(in.readMap((Writeable.Reader<String>) StreamInput::readString,
-                (Writeable.Reader<Tuple<Long, ElasticsearchException>>) in1 -> new Tuple<>(in1.readZLong(), in1.readException())));
-        if (in.getVersion().onOrAfter(Version.V_6_6_0)) {
-            autoFollowedClusters = new TreeMap<>(in.readMap(StreamInput::readString, AutoFollowedCluster::new));
-        } else {
-            autoFollowedClusters = Collections.emptyNavigableMap();
-        }
+        recentAutoFollowErrors = new TreeMap<>(in.readMap(StreamInput::readString,
+            in1 -> new Tuple<>(in1.readZLong(), in1.readException())));
+        autoFollowedClusters = new TreeMap<>(in.readMap(StreamInput::readString, AutoFollowedCluster::new));
```

Comment:
can you retain the comment and the cast?

Reply:
thanks for the catch! I must have deleted the comment initially, and then forgot why the cast was there, and then removed the cast. I've returned both.